scrapers/scraper.py

```diff
@@ -0,0 +1,267 @@
+   To change languages, change the languages, iso3to2 and family variables
+"""
+
+from collections import Counter
+import xml.etree.ElementTree as etree
+import argparse, re, sys, logging, os, json, requests, subprocess, shutil, stat
+from lexccounter import countStems as countLexcStems
+from dixcounter import get_info as countDixStems
+
+languages = [
+    'aze',
+    'bak',
+    'chv',
+    'crh',
+    'kaa',
+    'kaz',
+    'kir',
+    'kjh',
+    'kum',
+    'nog',
+    'sah',
+    'tat',
+    'tuk',
+    'tur',
+    'tyv',
+    'uig',
+    'uzb'
+    ]
+iso3to2 = {
+    'aze': 'az',
+    'bak': 'ba',
+    'chv': 'cv',
+    'crh': 'crh',
+    'kaa': 'kaa',
+    'kaz': 'kk',
+    'kir': 'ky',
+    'kjh': 'kjh',
+    'kum': 'kum',
+    'nog': 'nog',
+    'sah': 'sah',
+    'tat': 'tt',
+    'tuk': 'tk',
+    'tyv': 'tyv',
+    'tur': 'tr',
+    'uig': 'ug',
+    'uzb': 'uz'
+    }
+pairLocations = ["incubator", "nursery", "staging", "trunk"]
+langLocations = ["languages", "incubator"]
+states = ["prototype", "development", "working", "production"]
+family = "Turkic"
+wikiURL = "http://wiki.apertium.org/wiki/" + family + "_languages"
+
+def del_rw(action, name, exc):
+    """Force deletes file if access is denied"""
+    os.chmod(name, stat.S_IWRITE)
+    os.remove(name)
+
+def monoHistory(language):
+    """Returns the history of a monolingual dictionary"""
+    try:
+        oldFile = json.loads(open("../json/{}.json".format(language), "r").read())
+        for data in oldFile:
+            if data["name"] == language:
+                history = data["history"]
+                break
+            history = []
+    except FileNotFoundError:
+        history = []
+
+    subprocess.call(['git', 'clone', 'https://github.com/apertium/apertium-{}'.format(language)], stderr=subprocess.DEVNULL)
+    shutil.copyfile("./.mailmap", "./apertium-{}/.mailmap".format(language))
+    commits = subprocess.check_output(['git', 'log', '--use-mailmap', '--follow', 'apertium-{0}.{0}.lexc'.format(language)], cwd='./apertium-{}/'.format(language)).decode('utf-8').split('\n\n')
+    shutil.rmtree("./apertium-{}/".format(language), onerror=del_rw)
+
+    for commit in commits:
+        if commit.startswith(' '):
+            continue
+        commitData = {}
+        lines = commit.split('\n')
+        for line in lines:
+            if line.startswith('commit'):
+                commitData["sha"] = line[len("commit "):]
+            if line.startswith('Author'):
+                commitData["author"] = line.split(": ")[1].split(" <")[0]
+            if line.startswith('Date'):
+                commitData["date"] = line[len("date:"):].strip()
+        if any(commitData["sha"] == cm["sha"] for cm in history):
+            continue
+
+        lexcFile = requests.get("https://raw.githubusercontent.com/apertium/apertium-{0}/{1}/apertium-{0}.{0}.lexc".format(language, commitData["sha"])).text
+        if lexcFile == "404: Not Found\n":
```

Comment:
It should be possible to check the response status instead. I wouldn't rely on this text.